https://www.virtualbox.org/pipermail/vbox-dev/2023-September/010224.html

From: vbox-dev <vbox-dev-bounces at virtualbox.org> on behalf of Larry Finger via vbox-dev <vbox-dev at virtualbox.org>
Sent: Friday, August 11, 2023 12:23 AM
To: vbox-dev at virtualbox.org
Subject: [vbox-dev] Fix for kernel 6.5 build of VB 7.0.10

Hi,

The 7.0.10 release fixed most problems with kernel 6.5. The first exception was
the problem with extensible arrays in the kernel's vboxsf module that I
described earlier. There were some hiccups, but that fix should be in kernel
6.5-rc6.

One other problem showed up in the compilation of vbox_fb.c. The patch for that,
released under the MIT license, is shown below:

Index: VirtualBox-7.0.10/src/VBox/Additions/linux/drm/vbox_fb.c
===================================================================
--- a/src/VBox/Additions/linux/drm/vbox_fb.c
+++ b/src/VBox/Additions/linux/drm/vbox_fb.c
@@ -196,9 +196,13 @@
 	.owner = THIS_MODULE,
 	.fb_check_var = drm_fb_helper_check_var,
 	.fb_set_par = drm_fb_helper_set_par,
+#if RTLNX_VER_MIN(6, 5, 0)
+	FB_DEFAULT_SYS_OPS,
+#else
 	.fb_fillrect = drm_fb_helper_sys_fillrect,
 	.fb_copyarea = drm_fb_helper_sys_copyarea,
 	.fb_imageblit = drm_fb_helper_sys_imageblit,
+#endif
 	.fb_pan_display = drm_fb_helper_pan_display,
 	.fb_blank = drm_fb_helper_blank,
 	.fb_setcmap = drm_fb_helper_setcmap,
